CitationStyleLanguageSettingsForm.inc.php

```diff
@@ -47,27 +48,41 @@ public function initData()
         $this->setData('enabledCitationStyles', array_keys($this->plugin->getEnabledCitationStyles($contextId)));
         $this->setData('enabledCitationDownloads', $this->plugin->getEnabledCitationDownloads($contextId));
         $this->setData('publisherLocation', $this->plugin->getSetting($contextId, 'publisherLocation'));
+        $this->setData('groupAuthor', $this->plugin->getAuthorGroup($contextId));
+        $this->setData('groupTranslator', $this->plugin->getTranslatorGroup($contextId));
+        if ($this->plugin->isApplicationOmp()) {
+            $this->setData('groupEditor', $this->plugin->getEditorGroup($contextId));
+            $this->setData('groupChapterAuthor', $this->plugin->getChapterAuthorGroup($contextId));
+        }
     }
 
     /**
      * Assign form data to user-submitted data.
      */
-    public function readInputData()
+    public function readInputData(): void
     {
         $this->readUserVars([
             'primaryCitationStyle',
             'enabledCitationStyles',
             'enabledCitationDownloads',
             'publisherLocation',
+            'groupAuthor',
+            'groupTranslator'
         ]);
+        if ($this->plugin->isApplicationOmp()) {
+            $this->readUserVars(['groupEditor']);
+            $this->readUserVars(['groupChapterAuthor']);
+        }
     }
 
     /**
      * @copydoc Form::fetch()
      *
      * @param null|mixed $template
+     *
+     * @throws Exception
```

Comment:
No, my IDE told me that, because line 101 `$userGroupDao = DAORegistry::getDAO('UserGroupDAO');` can throw an exception. See here:

```
/**
     * Retrieve a reference to the specified DAO.
     *
     * @param $name string the class name of the requested DAO
     *
     * @return DAO
     */
    public static function &getDAO($name)
    {
        $daos = & DAORegistry::getDAOs();
        if (!isset($daos[$name])) {
            // Import the required DAO class.
            $application = \APP\core\Application::get();
            $className = $application->getQualifiedDAOName($name);
            if (!$className) {
                throw new \Exception('Unrecognized DAO ' . $name . '!');
            }

            // Only instantiate each class of DAO a single time
            $daos[$name] = new $className();
        }

        return $daos[$name];
    }
```

I also could use a try-catch-statment or we can say this case will never happen for 'UserGroupDAO' and ignore the possible exception. Which way you would prefer?